packages/core/src/utils/paths.ts

```diff
@@ -200,3 +200,28 @@ export function getUserCommandsDir(): string {
 export function getProjectCommandsDir(projectRoot: string): string {
   return path.join(projectRoot, GEMINI_DIR, COMMANDS_DIR_NAME);
 }
+
+/**
+ * Checks if a path is a subpath of another path.
+ * @param parentPath The parent path.
+ * @param childPath The child path.
+ * @returns True if childPath is a subpath of parentPath, false otherwise.
+ */
+export function isSubpath(parentPath: string, childPath: string): boolean {
+  // Heuristic to detect Windows paths even on POSIX systems
```

Comment:
is this heuristic for windows paths needed? Concerned we will make this code fragile if we sometimes require this heuristic and sometimes don't. would be preferred if we detect that the operating system is windows and special case based on that rather than pattern matching specific paths.